### PR TITLE
feat(generate-event): Add Amazon Lex V2 Event

### PR DESCRIPTION
**Pull request by michaelbrewer**
#### Which issue(s) does this change fix?

-  #3648

#### Why is this change necessary?

When working with Amazon Lex V2, it would be nice to have a way to generate sample events

#### How does it address the issue?

By adding a template for Amazon Lex V2

#### What side effects does this change have?

- Docs will also need to be updated: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-local-generate-event.html

#### Checklist

- [X] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [X] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [X] Write unit tests
- [X] Write/update functional tests
- [X] Write/update integration tests
- [X] `make pr` passes
- [X] `make update-reproducible-reqs` if dependencies were changed
- [X] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    "PickUpDate": null,
+                    "DriverAge": null,
+                    "CarType": null,
+                    "PickUpCity": null
+                },
+                "confirmationState": "None",
+                "name": "BookCar",
+                "state": "InProgress"
+            },
+            "nluConfidence": 1
```
Comment (moelasmar) [inline samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json:18]: I think nluConfidence should be in this format
```
           "nluConfidence": {
                "score": number
            },
```

File: samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            "confirmationState": "None",
+            "name": "BookCar",
+            "state": "InProgress"
+        },
+        "dialogAction": {
+            "slotToElicit": "PickUpCity",
+            "type": "ElicitSlot"
+        }
+    },
+    "responseContentType": "text/plain; charset=utf-8",
```
Comment (moelasmar) [inline samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json:61]: I believe the `responseContentType` should be in the following format:
```
"responseContentType": "CustomPayload | ImageResponseCard | PlainText | SSML",
```

File: samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    "messageVersion": "1.0",
+    "invocationSource": "DialogCodeHook",
+    "transcriptions": [
+        {
+            "resolvedSlots": {},
+            "resolvedContext": {
+                "intent": "BookCar"
+            },
+            "transcription": "Book a car",
+            "transcriptionConfidence": 1
```
Comment (moelasmar) [inline samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json:86]: I believe the `transcriptionConfidence` should be in the following format:
```
            "transcriptionConfidence": {
                "score": "number"
            },
```

File: samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        }
+    ],
+    "inputMode": "Text",
+    "bot": {
```
Comment (moelasmar) [inline samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json:90]: I think there is no `aliasName` in `bot` object, it should be in the following format
```
    "bot": {
        "id": "string",

        "name": "string",
        "aliasId": "string",
        "localeId": "string",
        "version": "string"
    },
```

File: samcli/lib/generated_sample_events/events/lex-v2/LexBookHotel.json [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            "resolvedSlots": {},
+            "resolvedContext": {
+                "intent": "BookHotel"
+            },
+            "transcription": "Book a hotel",
+            "transcriptionConfidence": 1
+        }
+    ],
+    "inputMode": "Text",
+    "bot": {
```
Comment (moelasmar) [inline samcli/lib/generated_sample_events/events/lex-v2/LexBookHotel.json:88]: I think there is no `aliasName` in `bot` object, it should be in the following format
```
    "bot": {
        "id": "string",

        "name": "string",
        "aliasId": "string",
        "localeId": "string",
        "version": "string"
    },
```

**Comment by michaelbrewer** [comment]
@moelasmar - weirdly, this is an actual recorded event from Lex V2. I will also look at the docs to see the disconnect :)

File: samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json (hunk 1 above)
Comment (michaelbrewer) [reply samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json:18]: Not true (i am afraid), these are actual sampled events.

File: samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json (hunk 2 above)
Comment (michaelbrewer) [reply samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json:61]: Not for a chat bot when testing

File: samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json (hunk 3 above)
Comment (michaelbrewer) [reply samcli/lib/generated_sample_events/events/lex-v2/LexBookCar.json:86]: Same here, this is a real sampled event

File: samcli/lib/generated_sample_events/events/lex-v2/LexBookHotel.json [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                    "RoomType": null,
+                    "CheckInDate": null,
+                    "Nights": null,
+                    "Location": null
+                },
+                "confirmationState": "None",
+                "name": "BookHotel",
+                "state": "InProgress"
+            },
+            "nluConfidence": 1
```
[the remark this answers is not in the document]
Comment (michaelbrewer) [reply samcli/lib/generated_sample_events/events/lex-v2/LexBookHotel.json:17]: Not when testing

**Comment by michaelbrewer** [comment]
@moelasmar - i  captured these when testing via the aws console and the Banking Bot examples:

- https://aws.amazon.com/getting-started/hands-on/create-banking-bot-on-amazon-lex-v2-console/

**Comment by michaelbrewer** [comment]
@moelasmar - this is also consistent with https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/aws-lambda/trigger/lex-v2.d.ts

**Review by moelasmar — APPROVED** [review]

**Review by mingkun2020 — APPROVED** [review]

**Comment by torresxb1** [comment]
Thanks for your contribution @michaelbrewer !

**Pull request merged**